Modules/_asynciomodule.c

```diff
@@ -434,15 +434,33 @@ future_schedule_callbacks(asyncio_state *state, FutureObj *fut)
         return 0;
     }
 
+    if (!PyList_CheckExact(fut->fut_callbacks)) {
```

Comment:
Is there a test for this? Is there a way to assign a non-list to fut._callbacks?

Reply:
Mmh, actually no. The property is read-only and even with subclassing it's not possible. I'll replace the list checks by assertions.